src/StructuredOutput/ChainProcessor.php

```diff
@@ -16,7 +16,7 @@ final class ChainProcessor implements InputProcessor, OutputProcessor
     private string $outputStructure;
 
     public function __construct(
-        private readonly ResponseFormatFactory $responseFormatFactory,
+        private readonly ResponseFormatFactoryInterface $responseFormatFactory,
```

Comment:
I added an interface to be able to test the ChainProcessor, maybe I am wrong, can you have a look?

Reply:
If you agree, I will move this to an extra PR

Reply:
gave it a look and another thought, go ahead - best option as well. if someone wants to fiddle around with the ResponseFactory i would rather enforce them to use composition over inheritance . classic :+1:

Reply:
Done in 
* #93